src/GenFacades/Program.cs

```diff
@@ -485,17 +485,24 @@ private void AddTypeForward(Assembly assembly, INamedTypeDefinition seedType)
                     assembly.ExportedTypes = new List<IAliasForType>();
                 assembly.ExportedTypes.Add(alias);
 
-                // Recursively add forwarders for all nested types regardless of their accessibility. This is
-                // how the C# compiler emits type forwarders for nested types. We might not need them for
-                // nested types that are not visible outside the assembly, but it is safer to replicate how
-                // the C# compiler works. Plus, it helps when diffing the output from ildasm for facades
-                // that were built from source vs. those that were produced by this tool.
-                //
+                // Recursively add forwarders for nested types.
                 // NOTE: Some design-time tools can resolve forwarded nested types with only the top-level forwarder,
                 //       but the runtime currently throws a TypeLoadException without explicit forwarders for the nested 
                 //       types.
-                foreach (var nestedType in seedType.NestedTypes.OrderBy(t => t.Name.Value))
-                    AddTypeForward(assembly, nestedType);
+                INamedTypeDefinition contractType = contractAssembly.GetAllTypes().Where(t => t.FullName() == seedType.FullName()).FirstOrDefault();
+                if (contractType != null) // Only add forwarders for the nested types on the contract
+                {
+                    foreach (var nestedType in contractType.NestedTypes.OrderBy(t => t.Name.Value))
```

Comment:
In fact we can actually create a table for the contract assembly that contains the nested type docids as well and we can align them, the same way we are aligning the other docids. That way we can eliminate this special casing and we can do the nested types the same as we do the non-nested types. All we have to do is ensure we traverse the nested types when building the docid lists above.